packages/passport/sdk/src/Passport.ts

```diff
@@ -198,7 +200,11 @@ export class Passport {
         logger.warn('Failed to retrieve a cached user session', error);
       }
       if (!user && !useCachedSession) {
-        user = await this.authManager.login(options?.anonymousId);
+        if (silent) {
+          user = await this.authManager.forceUserRefresh();
+        } else {
+          user = await this.authManager.login(options?.anonymousId);
+        }
```

Comment:
I wonder if it makes sense to do this check before calling `login` 🤔 That way, consumers can use `useCachedSession` & `useSilentLogin` to decide whether or not they want to fallback to displaying the login popup if the silent sign-in fails. E.g:
```
// ...
if (!user && useSilentLogin`) {
  user = await this.authManager.forceUserRefresh();
}

if (!user && !useCachedSession) {
  user = await this.authManager.login(options?.anonymousId);
}
// ...
```